pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -534,6 +546,35 @@ func (f *framework) GetWaitingPod(uid types.UID) WaitingPod {
 	return f.waitingPods.get(uid)
 }
 
+// ListPlugins returns a map of extension point name to plugin names
+// configured at each extension point.
+func (f *framework) ListPlugins() map[string][]string {
+	namesMap := make(map[string][]string)
+	namesMap["QueueSort"] = listPlugins(&f.queueSortPlugins)
+	namesMap["PreFilter"] = listPlugins(&f.preFilterPlugins)
+	namesMap["Filter"] = listPlugins(&f.filterPlugins)
+	namesMap["PostFilter"] = listPlugins(&f.postFilterPlugins)
+	namesMap["Score"] = listPlugins(&f.scorePlugins)
+	namesMap["Reserve"] = listPlugins(&f.reservePlugins)
+	namesMap["Premit"] = listPlugins(&f.permitPlugins)
+	namesMap["PreBind"] = listPlugins(&f.preBindPlugins)
+	namesMap["Bind"] = listPlugins(&f.bindPlugins)
+	namesMap["PostBind"] = listPlugins(&f.postBindPlugins)
+	namesMap["Unreserve"] = listPlugins(&f.unreservePlugins)
+	return namesMap
+}
+
+func listPlugins(pluginsPtr interface{}) []string {
+	plugins := reflect.ValueOf(pluginsPtr).Elem()
+	var names []string
+	for i := 0; i < plugins.Len(); i++ {
+		p := plugins.Index(i).Convert(reflect.TypeOf((*Plugin)(nil)).Elem())
```

Comment:
this conversion's only purpose is to panic if the element does not implement the Plugin interface. Is that what you want?
I find it unnecessary given that this function is unexported.

Reply:
no, I need the conversion to be able to call the method Name()

Reply:
But that's already done in the next line. Calling `Convert` is unnecessary.

Reply:
done.